core/pkg/leveldb/record.go

```diff
@@ -222,6 +222,18 @@ func (r *Reader) nextChunk(wantFirst bool) error {
 	}
 }
 
+// ReadHeader reads the first block and copies the header into a buffer
+func (r *Reader) ReadHeader(headerBuffer []byte) error {
+	l := len(headerBuffer)
+	n, err := io.ReadFull(r.r, r.buf[:])
+	if err != nil && err != io.ErrUnexpectedEOF {
```

Comment:
why explicitly check for io.ErrUnexpectedEOF? what if it is that?

Reply:
Because ReadFull reads a full 32kb block... so it might hit eof.   This was taken from the reader Next() pattern.

This is the stuff that is kinda ugly, but it felt best to add a bit of header functionality to leveldb rather than messing with internals on the initialization (i tried both ways)